source/common/unified_memory_allocation/include/uma/memory_provider.h

```diff
@@ -12,6 +12,31 @@
 #include <uma/base.h>
 #include <uma/memory_provider_ops.h>
 
+#define UMA_ATTR_NORMAL      0     /* No further special treatment.  */
+#define UMA_ATTR_RANDOM      1     /* Expect random page references.  */
+#define UMA_ATTR_SEQUENTIAL  2     /* Expect sequential page references.  */
+#define UMA_ATTR_WILLNEED    3     /* Will need these pages.  */
+#define UMA_ATTR_DONTNEED    4     /* Don't need these pages.  */
+#define UMA_ATTR_FREE        8     /* Free pages only if memory pressure.  */
+#define UMA_ATTR_REMOVE      9     /* Remove these pages and resources.  */
+#define UMA_ATTR_DONTFORK    10    /* Do not inherit across fork.  */
+#define UMA_ATTR_DOFORK      11    /* Do inherit across fork.  */
+#define UMA_ATTR_MERGEABLE   12    /* KSM may merge identical pages.  */
+#define UMA_ATTR_UNMERGEABLE 13    /* KSM may not merge identical pages.  */
+#define UMA_ATTR_HUGEPAGE    14    /* Worth backing with hugepages.  */
+#define UMA_ATTR_NOHUGEPAGE  15    /* Not worth backing with hugepages.  */
+#define UMA_ATTR_DONTDUMP    16    /* Explicity exclude from the core dump,
+                                      overrides the coredump filter bits.  */
+#define UMA_ATTR_DODUMP      17    /* Clear the UMA_ATTR_DONTDUMP flag.  */
+#define UMA_ATTR_WIPEONFORK  18    /* Zero memory on fork, child only.  */
+#define UMA_ATTR_KEEPONFORK  19    /* Undo UMA_ATTR_WIPEONFORK.  */
+#define UMA_ATTR_COLD        20    /* Deactivate these pages.  */
+#define UMA_ATTR_PAGEOUT     21    /* Reclaim these pages.  */
+#define UMA_ATTR_POPULATE_READ 22  /* Populate (prefault) page tables
+                                      readable.  */
+#define UMA_ATTR_POPULATE_WRITE 23 /* Populate (prefault) page tables
+                                      writable.  */
```

Comment:
I'm not sure about the intended use of those attributes, but assuming some of the could be used in conjunction we should define those using bit shift `(1 << #)`.

Reply:
Also, there's a gap between `4` and `8`.

Reply:
> I'm not sure about the intended use of those attributes, but assuming some of the could be used in conjunction we should define those using bit shift `(1 << #)`.

This assumption is wrong - they will not be used in conjunction.
They are defined exactly like the `MADV_*` flags:
https://sourceware.org/git/?p=glibc.git;a=blob;f=sysdeps/unix/sysv/linux/bits/mman-linux.h;h=6771f07a0f9a617fcde0eb264d54784788d22e76;hb=HEAD#l68